### PR TITLE
Fix consensus state vote summary deserialization

### LINKED ISSUE #836: Intermittent failure in CI when deserializing consensus state vote summaries
The most recent example of this is here: https://github.com/informalsystems/tendermint-rs/pull/834/checks?check_run_id=2192937025

I've managed to figure out that sometimes the vote type is encoded as a number (`1` or `2`), and sometimes it's encoded as a string (`SIGNED_MSG_TYPE_PREVOTE`), and the code currently only caters for the former.

**Steps to reproduce**

Try deserializing the string causing the CI failure:

```
Vote{0:2DA21E474F57 384/00/SIGNED_MSG_TYPE_PREVOTE(Prevote) 8FA9FD23F590 2987C33E8F87 @ 2021-03-25T12:12:03.693870115Z}
```

**What's the definition of "done" for this issue?**

When we can parse these kinds of vote summaries.

### PR DESCRIPTION
**Pull request by thanethomson**
Closes #836 

* [x] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [x] Wrote tests
* [x] Updated CHANGELOG.md

### CODE REVIEW

File: rpc/src/endpoint/consensus_state.rs [hunk 1]
```diff
@@ -384,4 +389,11 @@ mod test {
         };
         assert_eq!(summary.to_string(), "Vote{0:000001E443FD 1262197/00/1(Prevote) 634ADAF1F402 7BB974E1BA40 @ 2019-08-01T11:52:35.513572509Z}");
     }
+
+    #[test]
+    fn deserialization_regressions() {
+        for s in DESERIALIZATION_REGRESSIONS {
+            let _ = VoteSummary::from_str(s).unwrap();
```
Comment (romac) [inline rpc/src/endpoint/consensus_state.rs:396]: Can we test here that the parse `VoteSummary` contains the expected data?

**Review by romac — CHANGES_REQUESTED** [review]
Good stuff :)

File: rpc/src/endpoint/consensus_state.rs (hunk 1 above)
Comment (thanethomson) [reply rpc/src/endpoint/consensus_state.rs:396]: Sure! :+1:

**Review by shonfeder — APPROVED** [review]
Looks like the requested change her was addressed, so I'll provide the green check :)

**Review by romac — APPROVED** [review]
Looks good!

We might want to revisit this code altogether at some point, perhaps to rewrite it to use parser combinators or something like that as the format is a bit too involved for manual parsing. But in the meantime, with the new tests, we have some confidence that this works as expected and can refactor later.

**Comment by thanethomson** [comment]
> We might want to revisit this code altogether at some point, perhaps to rewrite it to use parser combinators or something like that as the format is a bit too involved for manual parsing. But in the meantime, with the new tests, we have some confidence that this works as expected and can refactor later.

Ideally we'd avoid this kind of custom serialization format altogether from the server side :grin: It should be something that's encoded properly in JSON/Protobuf, and not a custom string format.

**Comment by romac** [comment]
> Ideally we'd avoid this kind of custom serialization format altogether from the server side 😁 It should be something that's encoded properly in JSON/Protobuf, and not a custom string format.

💯

**Pull request merged**